core/src/main/scala/com/metamx/tranquility/config/ConfigHelper.scala

```diff
@@ -0,0 +1,114 @@
+package com.metamx.tranquility.config
+
+import java.io.InputStream
+import java.util.Properties
+
+import com.metamx.common.scala.Predef._
+import com.metamx.common.scala.Yaml
+import com.metamx.common.scala.untyped._
+import com.metamx.tranquility.druid.DruidBeams
+import com.metamx.tranquility.druid.DruidEnvironment
+import com.metamx.tranquility.druid.DruidGuicer
+import com.metamx.tranquility.druid.DruidLocation
+import com.metamx.tranquility.finagle.FinagleRegistry
+import com.metamx.tranquility.tranquilizer.Tranquilizer
+import io.druid.segment.realtime.FireDepartment
+import org.apache.curator.framework.CuratorFramework
+import org.skife.config.ConfigurationObjectFactory
+
+/**
+ * Helper methods to perform shared tasks based on common configuration file semantics.
+ */
+object ConfigHelper
+{
+  def readConfigYaml[T <: TranquilityConfig](
+    in: InputStream,
+    clazz: Class[T]
+    ): (T, Map[String, DataSourceConfig[T]], Dict) =
+  {
+    val yaml = in.withFinally(_.close()) { in =>
+      dict(Yaml.load(in))
+    }
+    val globalProperties: Dict = getAsDict(yaml, "properties")
+    val globalConfig = mapConfig(globalProperties, clazz)
+
+    for (prop <- globalProperties.filterKeys { _ startsWith("druid.extensions.") }) {
+      sys.props += (prop._1 -> prop._2.toString)
```

Comment:
Also a comment here indicating why we need to modify system properties (presumably because we need those set in order for the DruidGuicer to pick up extensions)